src/plugins/data/server/search/strategies/ese_search/ese_search_strategy.ts

```diff
@@ -82,10 +78,11 @@ export const enhancedEsSearchStrategyProvider = (
       return toAsyncKibanaSearchResponse({ ...body, response }, headers?.warning);
     };
 
-    const cancel = async () => {
-      if (id) {
-        await cancelAsyncSearch(id, esClient);
-      }
+    const cancel = () => {
+      if (!id || options.isStored) return;
+      cancelAsyncSearch(id, esClient).catch(() => {
+        // Swallow errors from cancellation
```

Comment:
These `cancel` methods end up being registered as event listeners in `pollSearch`, and as such should not throw errors (or else they will be unhandled), so we swallow them.